python/lsst/resources/_resourcePath.py

```diff
@@ -1103,11 +1129,41 @@ def walk(
         """
         raise NotImplementedError()
 
+    @overload
+    @classmethod
+    def findFileResources(
+        cls,
+        candidates: Iterable[ResourcePathExpression],
+        file_filter: Optional[Union[str, re.Pattern]],
+        grouped: Literal[True],
+    ) -> Iterator[Iterator[ResourcePath]]:
+        pass
+
+    @overload
     @classmethod
     def findFileResources(
         cls,
-        candidates: Iterable[Union[str, ResourcePath]],
-        file_filter: Optional[str] = None,
+        candidates: Iterable[ResourcePathExpression],
+        *,
+        grouped: Literal[True],
+    ) -> Iterator[Iterator[ResourcePath]]:
+        pass
+
+    @overload
+    @classmethod
+    def findFileResources(
+        cls,
+        candidates: Iterable[ResourcePathExpression],
+        file_filter: Optional[Union[str, re.Pattern]] = None,
+        grouped: Literal[False] = False,
+    ) -> Iterator[ResourcePath]:
+        pass
+
+    @classmethod
+    def findFileResources(
+        cls,
+        candidates: Iterable[ResourcePathExpression],
+        file_filter: Optional[Union[str, re.Pattern]] = None,
         grouped: bool = False,
     ) -> Iterator[Union[ResourcePath, Iterator[ResourcePath]]]:
```

Comment:
Does mypy need these type annotations if the overloads exist?

Reply:
Frustratingly, it complains if I remove them entirely.  So it's either this or replacing them all with `Any`, and I think this is slightly better, in part because we've already got it.